src/proto/network/mod.rs

```diff
@@ -0,0 +1,34 @@
+//! Network access protocols.
+//!
+//! These protocols can be used to interact with network resources.
+
+pub mod pxe;
+
+/// Represents an IPv4/v6 address. Corresponds to `EFI_IP_ADDRESS` type in the
+/// C API.
+#[derive(Clone, Copy)]
```

Comment:
I'd suggest also adding the `PartialEq` and `Eq` traits, since it might be useful to compare IP addresses for equality. Not sure, but maybe also `Hash`?